File: 0299-bulls-and-cows/0299-bulls-and-cows.java
class Solution {
    public String getHint(String secret, String guess) {
        int[] map = new int[10];
        int bulls = 0;
        for (int i = 0; i < secret.length(); i++) {
            if (guess.charAt(i) != secret.charAt(i))
                map[guess.charAt(i) - '0']++;
            else
                bulls++;
        }
        int cows = 0;
        for (int i = 0; i < secret.length(); i++) {
            if (guess.charAt(i) != secret.charAt(i)) {
                if (map[secret.charAt(i) - '0'] > 0) {
                    cows++;
                    map[secret.charAt(i) - '0']--;
                }
            }
        }
        String ans = bulls + "A" + cows + "B";
        return ans; 
    }
}

File: 0299-bulls-and-cows/0299-bulls-and-cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
         vector<int>map(10,0);
        int bulls=0;
        for(int i=0;i<secret.size();i++){
            if(guess[i]!=secret[i])map[guess[i]-'0']++;
            else bulls++;
        }
        int cows=0;
        for(int i=0;i<secret.size();i++){
            if(guess[i]!=secret[i]){
                if(map[secret[i]-'0']>0){
                    cows++;
                    map[secret[i]-'0']--;
                }
            }
            
        }
        string ans=to_string(bulls)+"A"+to_string(cows)+"B";
        return ans;
    }
};